### PR TITLE
fix: use sfu api key if present

### LINKED ISSUE #90: provide API-key to SFU management endpoints
Add the api-key header `X-APIkey` with a configurable API-key when communicating with the SFU management endpoint

### PR DESCRIPTION
**Pull request by marcusspangenberg**
Closes #90

### CODE REVIEW

**Review by birme — APPROVED** [review]
Looks good. Just add the env key to the server endpoint README.
Or even better make it an option to the WhipEndpoint constructor.

**Comment by marcusspangenberg** [comment]
> Looks good. Just add the env key to the server endpoint README. Or even better make it an option to the WhipEndpoint constructor.

Please have a look again, I moved all the mandatory setter calls to constructor parameters.

File: packages/server/src/whip/sfu/sfuWhipResource.ts [hunk 1]
```diff
@@ -383,22 +392,6 @@ export class SfuWhipResource implements WhipResource {
     return Promise.resolve(this.answer);
   }
 
-  assignBroadcasterClients(broadcasterClientSfuPairs: BroadcasterClientSfuPair[]) {
```
Comment (birme) [inline packages/server/src/whip/sfu/sfuWhipResource.ts:386]: Can we keep these two functions but mark as deprecated?
There are already some examples and components out there that uses them.

**Review by birme — CHANGES_REQUESTED** [review]
Keep and deprecate the removed exported methods to assign SFU origin and broadcaster clients.

**Review by birme — APPROVED** [review]

**Pull request merged**